lib/parse/client.rb

```diff
@@ -69,6 +69,11 @@ def request(uri, method = :get, body = nil, query = nil, content_type = nil)
       end
 
       @session.send(method, uri, query || body || {}, headers).body
+
+    # Don't leak our internal libraries to our clients.
+    # Extend this list of exceptions as needed.
+    rescue Faraday::Error::ConnectionFailed, Faraday::Error::TimeoutError
+      raise Parse::ConnectionError
```

Comment:
Thanks! You do have a point (and I have just been bitten by this problem) but you should rescue the exception object too. 

- rescue the error and the object (rescue (x, y, z) => e)
- log the error (the client object has a `logger`)
- raise Parse::ConnectionError with the wrapped exception